lib/elixir/lib/system.ex

```diff
@@ -5,40 +5,71 @@ defmodule System do
   with the VM or the host system.
   """
 
-  defp strip_re(iodata, pattern) do
-    :re.replace(iodata, pattern, "", [return: :binary])
+  @base_dir           Path.join([__DIR__, "..", "..", ".."])
+  @version_file       Path.join(@base_dir, "VERSION")
+  @git_dir            Path.join(@base_dir, ".git")
+  @git_head_file      Path.join(@git_dir, "HEAD")
+
+  @spec strip(iodata) :: String.t
+  defp strip(iodata) do
+    iodata
+    |> :re.replace("^[\s\r\n\t]+", "", [return: :binary])
+    |> :re.replace("[\s\r\n\t]+$", "", [return: :binary])
   end
 
+  @spec read_stripped(String.t) :: String.t
   defp read_stripped(path) do
-    case :file.read_file(path) do
+    case File.read(path) do
       {:ok, binary} ->
-        strip_re(binary, "^\s+|\s+$")
-      _ -> ""
+        strip(binary)
+      _ ->
+        ""
     end
   end
 
   # Read and strip the version from the VERSION file.
+  @spec get_version :: String.t
   defmacrop get_version do
-    case read_stripped(:filename.join(__DIR__, "../../../VERSION")) do
+    case read_stripped(@version_file) do
       ""   -> raise RuntimeError, message: "could not read the version number from VERSION"
       data -> data
     end
   end
 
-  # Tries to run "git rev-parse --short HEAD". In the case of success returns
-  # the short revision hash. If that is not available, tries to read the commit hash
+  # Tries to run "git rev-parse HEAD". In the case of success returns
+  # the revision hash. If that is not available, tries to read the commit hash
   # from .git/HEAD. If that fails, returns an empty string.
+  @spec get_revision :: String.t
```

Comment:
because we don't know if users want to user the full version for something else, like linking to the source file in github...
It'd better give the full version and letting them shorten it, than having them to do hacky stuck such as calling `git  rev-parse HEAD`.
but it's up to you the final word